tools/contrib/fsurdat_modifier.py

```diff
@@ -0,0 +1,356 @@
+
+
+    @staticmethod
+    def update_metadata(nc, filename):
+
+        #update attributes
+        today = date.today()
+        today_string = today.strftime("%Y-%m-%d")
+
+        #get git hash
+        sha = get_git_sha()
+
+        nc.attrs['Created_on'] = today_string
+        nc.attrs['Created_by'] = myname
+        nc.attrs['Created_with'] = os.path.abspath(__file__) + " -- " + sha
+        nc.attrs['Created_from'] = filename
+
+        #delete unrelated attributes if they exist
+        del_attrs = ['source_code', 'SVN_url', 'hostname', 'history'
+                     'History_Log', 'Logname', 'Host', 'Version',
+                     'Compiler_Optimized']
+        attr_list = nc.attrs
+
+        for attr in del_attrs:
+            if attr in attr_list:
+                del(nc.attrs[attr])
+
+
+    def modify(self):
+
+        print ("Creating surface dataset file")
+        filename = self.fsurf_in
+        print("Open file: " + filename)
+        f1 = xr.open_dataset(filename)
+
+        # modify surface data properties
+        if self.overwrite_single_pft:
+            f1['PCT_NAT_PFT'][:,:,:] = 0
+            f1['PCT_NAT_PFT'][:,:,self.dominant_pft] = 100
+        if self.zero_nonveg_landunits:
+            f1['PCT_NATVEG'][:,:]  = 100
+            f1['PCT_CROP'][:,:]    = 0
+            f1['PCT_LAKE'][:,:]    = 0.
+            f1['PCT_WETLAND'][:,:] = 0.
+            f1['PCT_URBAN'][:,:,]   = 0.
+            f1['PCT_GLACIER'][:,:] = 0.
+        if self.uniform_snowpack:
+            f1['STD_ELEV'][:,:] = 20.
+        if self.no_saturation_excess:
+            f1['FMAX'][:,:] = 0.
+
+        # specify dimension order 
+        f1= f1.transpose(u'time', u'cft', u'lsmpft', u'natpft', u'nglcec', u'nglcecp1', u'nlevsoi', u'nlevurb', u'numrad', u'numurbl', 'lsmlat', 'lsmlon')
+        
+        #update attributes
+        self.update_metadata(f1, filename)
+
+        # mode 'w' overwrites file if it exists
+        f1.to_netcdf(path=self.fsurf_out, mode='w')
+        print('Successfully created file (fsurf_out) :' + self.fsurf_out)
+        f1.close()
+
+
+def setup_logging(log_file, log_level):
+    """
+    Setup logging to log to console and log file.
+    """
+
+    root_logger = logging.getLogger()
+    root_logger.setLevel(log_level)
+
+    # setup log file
+    one_mb = 1000000
+    handler = logging.handlers.RotatingFileHandler(log_file, maxBytes=one_mb , backupCount=10)
+
+    fmt = logging.Formatter(
+            '%(asctime)s %(name)-12s %(levelname)-8s %(message)s',
+            datefmt='%y-%m-%d %H:%M:%S')
+
+    handler.setFormatter(fmt)
+    root_logger.addHandler(handler)
+
+    # setup logging to console
+    stream_handler = logging.StreamHandler(sys.stdout)
+    stream_handler.setFormatter(fmt)
+    root_logger.addHandler(stream_handler)
+
+    # redirect stdout/err to log file
+    StreamToLogger.setup_stdout()
+    StreamToLogger.setup_stderr()
+
+
+class StreamToLogger(object):
```

Comment:
- [ ] Again, this should be moved to a general place. However, I'd like to have some discussion on what here we really want. At a glance, it looks like this is doing something like redirecting print statements to the logger. My feeling is that we shouldn't do that, but instead should explicitly make logger calls at the appropriate level (logger.info, logger.debug, logger.warning, etc.) because that lets the user set the desired logging level for the script.